File: app/src/main/java/me/hachy/routineweek/fragment/WeeklyFragment.java
package me.hachy.routineweek.fragment;


import android.os.Bundle;
import android.support.design.widget.TabLayout;
import android.support.v4.app.Fragment;
import android.support.v4.view.ViewPager;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import java.util.Calendar;
import java.util.TimeZone;

import me.hachy.routineweek.R;
import me.hachy.routineweek.adapter.WeeklyPagerAdapter;


public class WeeklyFragment extends Fragment {

    public WeeklyFragment() {
        // Required empty public constructor
    }

    public static WeeklyFragment newInstance() {
        return new WeeklyFragment();
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_weekly, container, false);

        TabLayout tabLayout = (TabLayout) view.findViewById(R.id.tab_layout);
        ViewPager viewpager = (ViewPager) view.findViewById(R.id.view_pager);

        Calendar cal = Calendar.getInstance(TimeZone.getDefault());
        int day = cal.get(Calendar.DAY_OF_WEEK);

        viewpager.setAdapter(new WeeklyPagerAdapter(getFragmentManager(), getActivity()));
        tabLayout.setupWithViewPager(viewpager);

        if (day == Calendar.MONDAY) {
            viewpager.setCurrentItem(0);
        } else if (day == Calendar.TUESDAY) {
            viewpager.setCurrentItem(1);
        } else if (day == Calendar.WEDNESDAY) {
            viewpager.setCurrentItem(2);
        } else if (day == Calendar.THURSDAY) {
            viewpager.setCurrentItem(3);
        } else if (day == Calendar.FRIDAY) {
            viewpager.setCurrentItem(4);
        } else if (day == Calendar.SATURDAY) {
            viewpager.setCurrentItem(5);
        } else if (day == Calendar.SUNDAY) {
            viewpager.setCurrentItem(6);
        }

        return view;
    }

}

File: app/src/main/java/me/hachy/routineweek/fragment/WeeklyFragment.kt
package me.hachy.routineweek.fragment


import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import kotlinx.android.synthetic.main.fragment_weekly.*

import java.util.Calendar
import java.util.TimeZone

import me.hachy.routineweek.R
import me.hachy.routineweek.adapter.WeeklyPagerAdapter


class WeeklyFragment : Fragment() {

    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater?.inflate(R.layout.fragment_weekly, container, false)
    }

    override fun onViewCreated(view: View?, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        val cal = Calendar.getInstance(TimeZone.getDefault())
        val day = cal.get(Calendar.DAY_OF_WEEK)

        viewPager.adapter = WeeklyPagerAdapter(fragmentManager, activity)
        tabLayout.setupWithViewPager(viewPager)

        when (day) {
            Calendar.MONDAY -> viewPager.currentItem = 0
            Calendar.TUESDAY -> viewPager.currentItem = 1
            Calendar.WEDNESDAY -> viewPager.currentItem = 2
            Calendar.THURSDAY -> viewPager.currentItem = 3
            Calendar.FRIDAY -> viewPager.currentItem = 4
            Calendar.SATURDAY -> viewPager.currentItem = 5
            Calendar.SUNDAY -> viewPager.currentItem = 6
        }
    }

    companion object {

        fun newInstance(): WeeklyFragment {
            return WeeklyFragment()
        }
    }

}// Required empty public constructor
